lib/src/connector/timings.cc

```diff
@@ -29,39 +70,87 @@ ConnectionTimings::Duration_us ConnectionTimings::getTCPInterval() const
         tcp_pre_init - start);
 }
 
-ConnectionTimings::Duration_us ConnectionTimings::getHandshakeInterval() const
+ConnectionTimings::Duration_us
+ConnectionTimings::getOpeningHandshakeInterval() const
 {
+    if (!_open)
+        return Duration_us::zero();
+
     return boost::chrono::duration_cast<ConnectionTimings::Duration_us>(
         tcp_post_init - tcp_pre_init);
 }
 
 ConnectionTimings::Duration_us ConnectionTimings::getWebSocketInterval() const
 {
+    if (!_open)
+        return Duration_us::zero();
+
     return boost::chrono::duration_cast<ConnectionTimings::Duration_us>(
         open - start);
 }
 
-ConnectionTimings::Duration_us ConnectionTimings::getCloseInterval() const
+ConnectionTimings::Duration_us
+ConnectionTimings::getClosingHandshakeInterval() const
 {
+    if (!_open || !_closing_started || !_closed)
```

Comment:
Do you need to test the `_open` here? I.e. Can it be in `_closing_started` & `_closed` without going through `_open`?